source/integrations/mongoose-get-started.txt

```diff
@@ -0,0 +1,666 @@
+
+- Set up your environment to use Mongoose
+- Connect to MongoDB
+- Create a Mongoose schema and model
+- Insert, update, find, and delete data
+- Project document fields
+- Validate your data
+- Use multiple schemas and middleware
+
+Prerequisites
+~~~~~~~~~~~~~
+
+Before you begin this tutorial, perform the following actions:
+
+- Set up a MongoDB Atlas account and configure a cluster.
+  To view instructions, see the :atlas:`Get Started with Atlas
+  </getting-started>` guide.
+- Install `Node.js <https://nodejs.org/en/download>`__ {+min-node-version+} or later.
+
+.. procedure::
+   :style: connected
+
+   .. step:: Set up your environment.
+
+      This tutorial uses `nodemon <https://www.npmjs.com/package/nodemon>`__ to
+      run the code locally. Run the following commands in your terminal to
+      initialize your project and install the necessary dependencies: 
+
+      .. code-block:: bash
+
+         mkdir mongodb-mongoose
+         cd mongodb-mongoose
+         npm init -y
+         npm i mongoose
+         npm i -D nodemon
+
+      Open your project in your preferred code editor. This tutorial uses ES
+      Modules instead of CommonJS. You must add the ``module`` type to use ES
+      Modules. Go to the ``package.json`` file and add the following code:
+
+      .. code-block:: json
+
+         ...
+         "scripts": {
+           "dev": "nodemon index.js"
+         },
+         "type": "module",
+         ...
+
+   .. step:: Connect to MongoDB.
+
+      In the root level of your project, create a file named ``index.js`` and
+      add the following code to the top of the file:
+
+      .. code-block:: javascript
+
+         import mongoose from 'mongoose';
+
+         mongoose.connect("<connection string>")
+
+      Replace the ``<connection string>`` placeholder with your MongoDB Atlas
+      connection string. For more information on how to find your connection
+      string, see the :atlas:`Connect to Your Cluster
+      </tutorial/connect-to-your-cluster>` tutorial in the Atlas documentation. 
+
+   .. step:: Create a schema and a model.
+
+      Before you start adding and updating data in MongoDB, you must create a
+      schema and model.
+
+      With Mongoose, you create a schema model file for each schema that is
+      needed. First, create the a folder called ``model`` to put all your schema
+      files in, then create your first schema file called ``Blog.js``. Open this
+      file and add the following code:  
+
+      .. literalinclude:: /includes/integrations/mongoose-blogSchema.js
+         :language: javascript
+         :dedent:
+
+   .. step:: Perform CRUD operations.
+
+      You now have your first model and schema set up, and you can start
+      inserting data into the database. The following sections show you how to
+      perform CRUD operations using Mongoose.
+
+      .. procedure::
+         :style: connected
+
+      Insert Data
+      ~~~~~~~~~~~
```

Comment:
This section down through the Delete Data section needs be indented further to be under the procedure directive - currently it isn't rendering on the page